test.sh

```diff
@@ -23,6 +23,11 @@ set -o xtrace
 # This script should work no matter where you call it from.
 cd "$(dirname "$0")"
 
+# Pre-commit checks are only runing on changed files.
+changed_files=$(git diff --name-only $TRAVIS_BRANCH...HEAD)
+echo "Changed files: " $changed_files
+pre-commit run --show-diff-on-failure --files $changed_files
```

Comment:
Hey, that's cool!

I could be wrong, but I *think* the clang-format check still won't do anything.  That would require (maybe?) `git reset $TRAVIS_BRANCH`, to uncommit all of the changes.  And then I don't think you'd need to specify `--files`.  I could be wrong, though; hopefully it's not too hard to test.

If you just want to run all of the *other* pre-commits on changed files, this works for me!  Perhaps add a comment expressing your intent either way?

Reply:
Done, thanks.